server/src/main/java/org/opensearch/index/mapper/ParametrizedFieldMapper.java

```diff
@@ -276,7 +282,7 @@ public Parameter<T> setValidator(Consumer<T> validator) {
         /**
          * Configure a custom serializer for this parameter
          */
-        public Parameter<T> setSerializer(Serializer<T> serializer, Function<T, String> conflictSerializer) {
+        protected Parameter<T> setSerializer(Serializer<T> serializer, Function<T, String> conflictSerializer) {
```

Comment:
@reta I saw this failed for k-NN plugin field mapper: https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/mapper/KNNVectorFieldMapper.java#L138. Should Serializer be public? It seems like it is similar to setValidator.

Reply:
Thanks @jmazanec15 , I will take a look shortly, the `Serializer` is defined as protected interface

```
    @InternalApi
    protected interface Serializer<T> {
        void serialize(XContentBuilder builder, String name, T value) throws IOException;
    }

```

so it was really weird to have a public method that uses the protected interface

Reply:
@jmazanec15 the change was reverted, should be all good, thank you for noticing!